src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/NotFoundFileInfo.cs

```diff
@@ -16,9 +16,9 @@ public class NotFoundFileInfo : IFileInfo
         /// Initializes an instance of <see cref="NotFoundFileInfo"/>.
         /// </summary>
         /// <param name="name">The name of the file that could not be found</param>
-        public NotFoundFileInfo(string name)
+        public NotFoundFileInfo(string? name)
```

Comment:
I'm not sure this is a good change. If you are constructing a `NotFoundFileInfo`, you should be passing the name in. Allowing `null` here is misleading to the usage of the class. Even though it doesn't throw an exception, we should still annotate it as this needs to be non-null.